src/partition/growfs.c

```diff
@@ -31,7 +33,8 @@ static bool arg_dry_run = false;
 
 #if HAVE_LIBCRYPTSETUP
 static int resize_crypt_luks_device(dev_t devno, const char *fstype, dev_t main_devno) {
-        _cleanup_free_ char *devpath = NULL, *main_devpath = NULL;
+        _cleanup_(sd_device_unrefp) sd_device *main_dev = NULL, *dev = NULL;
+        const char *devpath = NULL, *main_devpath = NULL;
```

Comment:
It is not necessary to initialize `devpath` and `main_devpath` with `NULL`. Please drop them.